src/test/kotlin/com/forgerock/sapi/gateway/ob/uk/support/payment/PaymentAS.kt

```diff
@@ -54,18 +56,35 @@ class PaymentAS : GeneralAS() {
         val cookie = "$AM_COOKIE_NAME=${response.tokenId}"
         val consentRequest = continueAuthorize(authorizeURL, cookie)
         val consentDetails = getConsentDetails(consentRequest, cookie)
-        val debtorAccount = getDebtorAccountFromConsentDetails(consentDetails)
-        val consentDecisionResponse = sendConsentDecision(consentRequest, debtorAccount, decision, cookie)
+        val consentedAccount = getConsentedAccountFromConsentDetails(consentDetails)
+        val isDebtorAccountProvided = isDebtorAccountProvided(consentDetails)
+        val consentDecisionResponse = sendConsentDecision(
+            consentRequest,
+            consentedAccount,
+            decision,
+            isDebtorAccountProvided,
+            cookie
+        )
         val authCode = getAuthCode(consentDecisionResponse.consentJwt, consentDecisionResponse.redirectUri, cookie)
         return exchangeCode(registrationResponse, tpp, authCode)
     }
 
-    private fun getDebtorAccountFromConsentDetails(consentDetails: String): FRFinancialAccount {
+    private fun isDebtorAccountProvided(consentDetails: String): Boolean {
+        val str = JsonParser.parseString(consentDetails).asJsonObject
+        val initiation = str.getAsJsonObject("initiation")
+        if (Objects.nonNull(initiation)) {
+            if (Objects.nonNull(initiation.getAsJsonObject("debtorAccount"))) {
+                return true
+            }
+        }
+        return false
+    }
+
+    private fun getConsentedAccountFromConsentDetails(consentDetails: String): FRFinancialAccount {
         try {
-            val str = JsonParser().parse(consentDetails).asJsonObject
+            val str = JsonParser.parseString(consentDetails).asJsonObject
```

Comment:
Why are we using this low level API rather than Gson.fromJson()?

Reply:
The `JsonParser() constructor instance` has been deprecated.
```java
  /** @deprecated No need to instantiate this class, use the static methods instead. */
  @Deprecated
  public JsonParser() {}
```